test/unit/nedap/utils/spec/api/check.cljc

```diff
@@ -0,0 +1,50 @@
+(ns unit.nedap.utils.spec.api.check
+  (:require
+   #?(:clj [clojure.spec.alpha :as spec] :cljs [cljs.spec.alpha :as spec])
+   #?(:clj [clojure.test :refer [deftest testing are is use-fixtures]] :cljs [cljs.test :refer-macros [deftest testing is are] :refer [use-fixtures]])
+   [nedap.utils.spec.api :as sut])
+  #?(:clj (:import (clojure.lang ExceptionInfo))))
+
+(spec/def ::number number?)
+
+(defn backed-by-dynamic-spec-checking
+  "A function that performs an `assert` using a dynamic spec.
+
+  This challenges our `check!` helper: it shouldn't report the `'spec` symbol as the relevant spec,
+  but the underlying spec value instead."
+  [spec x]
+  (assert (sut/check! spec x))
+  x)
+
+(def sample-spec-object (spec/spec #{1 2 3}))
+
+(deftest reported-ex-data
+
+  (are [passed-spec expected] (testing [passed-spec expected]
+                                (try
+                                  (with-out-str
+                                    (backed-by-dynamic-spec-checking passed-spec "NaN"))
+                                  (is false)
+                                  (catch ExceptionInfo e
```

Comment:
Should `ExceptionInfo` be `#?(:clj ExceptionInfo :cljs js/Error)` ?

Reply:
![image](https://user-images.githubusercontent.com/1162994/72413423-110ab280-3770-11ea-9e95-3ae363d245b9.png)